src/main/java/com/andersen/collectionzz/CustomHashSet.java

```diff
@@ -0,0 +1,35 @@
+package com.andersen.collectionzz;
+
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class CustomHashSet <T> {
+
+	private HashMap<T, Object> map;
+	private static final Object PRESENT = new Object();
+
+	public CustomHashSet() {
+		map = new HashMap<>();
+	}
+
+	public Iterator<T> iterator() {
+		return map.keySet().iterator();
+	}
+
+	public boolean contains(Object o) {
+		return map.containsKey(o);
+	}
+
+	public boolean add(T e) {
+		return map.put(e, PRESENT) == null;
```

Comment:
why == null?

Reply:
because otherwise it doesn't return boolean value, and hashset from jdk library has the same code, I can just change the return type to void

Reply:
Okay, or update to void, or leave is it is)